### PR TITLE
Backporting case-sensitivity flag to expression module

### PR DESCRIPTION
**Pull request by TGooch44**
This adds case-insensitive option to the expression module along with the associated tests.  This is the first in a series of PR's that will start paving the road for read support in the python client. @rdblue

### CODE REVIEW

File: python/iceberg/api/expressions/inclusive_metrics_evaluator.py [hunk 1]
```diff
@@ -52,13 +53,13 @@ def __init__(self, expr, schema, struct):
         self.struct = struct
 
     def eval(self, file):
-        if file.record_count <= 0:
+        if file.record_count() <= 0:
```
Comment (danielcweeks) [inline python/iceberg/api/expressions/inclusive_metrics_evaluator.py]: Are these changes related to the case sensitivity or just a bug?

File: python/iceberg/api/expressions/strict_metrics_evaluator.py [hunk 2]
```diff
@@ -51,13 +51,13 @@ def __init__(self, expr, schema, struct):
             self.upper_bounds = None
 
         def eval(self, file):
-            if file.record_count <= 0:
+            if file.record_count() <= 0:
```
Comment (danielcweeks) [inline python/iceberg/api/expressions/strict_metrics_evaluator.py]: Same here, looks unrelated to the case sensitivity.

File: python/iceberg/api/expressions/inclusive_metrics_evaluator.py (hunk 1 above)
Comment (TGooch44) [reply python/iceberg/api/expressions/inclusive_metrics_evaluator.py]: both items were/are bugs, that were fixed between the initial commit and when I started the case-sensitivity backport

Comment (danielcweeks) [reply python/iceberg/api/expressions/inclusive_metrics_evaluator.py]: ok, this is trivial, so I'll merge, but best to split those out and make smaller commits going forward (where possible).

**Comment by danielcweeks** [comment]
+1
